src/ui/connectiondialog.cc

```diff
@@ -231,18 +240,26 @@ void ConnectionDialog::saveSettings() {
       ui_->server_executable_line_edit->text());
   util::settings::connection::setShutDownServerOnQuit(
       ui_->shut_down_server_check_box->isChecked());
-}
 
-void ConnectionDialog::saveSettingsToggled(bool toggled) {
-  ui_->save_key_check_box->setEnabled(toggled);
```

Comment:
You removed this function which was called when removed `save_settings_check_box` was used, which causes that `save_key_check_box` is always disabled and unusable.